pennylane/debugging/debugger.py

```diff
@@ -132,33 +132,198 @@ def pldb_device_manager(device):
+    execution of the quantum circuit, an interactive debugging prompt is launched to step
+    throught the circuit execution using Pdb like commands (:code:`list`, :code:`next`,
+    :code:`continue`, :code:`quit`).
+
+    **Example**
+
+    Consider the following python script containing the quantum circuit with breakpoints.
+
+    .. code-block:: python3
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.breakpoint()
+
+            qml.RX(x, wires=0)
+            qml.Hadamard(wires=1)
+
+            qml.breakpoint()
+
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.Z(0))
+
+        circuit(1.23)
+
+    Running the above python script opens up the interactive :code:`[pldb]:` prompt in the terminal.
+    The prompt specifies the path to the script along with the next line to be executed after the breakpoint.
+
+    .. code-block:: console
+
+        > /Users/your/path/to/script.py(8)circuit()
+        -> qml.RX(x, wires=0)
+        [pldb]:
+
+    We can interact with the prompt using the commands: :code:`list` , :code:`next`,
+    :code:`continue`, and :code:`quit`. Additionally, we can also access any variables defined in the function.
+
+    .. code-block:: console
+
+        [pldb]: x
+        1.23
+
+    The :code:`list` command will print a section of code around the breakpoint, highlighting the next line
+    to be executed.
+
+    .. code-block:: console
+
+        [pldb]: list
+          3
+          4  	@qml.qnode(dev)
+          5  	def circuit(x):
+          6  	    qml.breakpoint()
+          7
+          8  ->	    qml.RX(x, wires=0)
+          9  	    qml.Hadamard(wires=1)
+         10
+         11  	    qml.breakpoint()
+         12
+         13  	    qml.CNOT(wires=[0, 1])
+        [pldb]:
+
+    The :code:`next` command will execute the next line of code, and print the new line to be executed.
+
+    .. code-block:: console
+
+        [pldb]: next
+        > /Users/your/path/to/script.py(9)circuit()
+        -> qml.Hadamard(wires=1)
+        [pldb]:
+
+    The :code:`continue` command will resume code execution until another breakpoint is reached. It will
+    then print the new line to be executed. Finally, :code:`quit` will resume execution of the file and
+    terminate the debugging prompt.
+
+    .. code-block:: console
+
+        [pldb]: continue
+        > /Users/your/path/to/script.py(13)circuit()
+        -> qml.CNOT(wires=[0, 1])
+        [pldb]: quit
+
+    """
     PLDB.valid_context()  # Ensure its being executed in a valid context
 
     debugger = PLDB(skip=["pennylane.*"])  # skip internals when stepping through trace
     debugger.set_trace(sys._getframe().f_back)  # pylint: disable=protected-access
 
 
-def state():
-    """Compute the state of the quantum circuit.
+def debug_state():
+    """Compute the quantum state at the current point in the quantum circuit.
 
```

Comment:
Might be worth mentioning that this does not alter the state in the rest of the circuit.

Reply:
Same for expval

Reply:
Added to the documentation for `state`, `expval` and `probs`.